### PR TITLE
Fix: Shorter command alternative for interacting with node-mongo CLI needed

### LINKED ISSUE #43: Add prompt for options.template in cli.js
## Intro
📌 **Note:** We are using `node-mongo` command for our cli project, not `create-project`.

* For our cli project, we have 3 templates - `es6`, `cjs`, `ts-es6`. These have been stored inside the `templateCollection` variable for you to use. 
* `es6` is the default template and has been stored in the `defaultTemplate` variable for you to use.
* The command line arguments and the order with which they are written in the command line have also been done for you. For example in the command below, `ts-folder` is the `folderName` i.e. `arg._[0]` while `ts-es6` is the template i.e. `arg._[1]`.
* A folder is automatically created from your entry. Since the template is specified and valid, it automatically downloads the template files too.

Run `node-mongo ts-folder ts-es6` to test/see output in console & the folder automatically created from your entry.

#### Output:
template-ts-es6-output

📌 **Hint:** The only place you should be writing code for the template prompt task is the `promptForMissingOptions` function inside the [cli.js file](https://github.com/code-collabo/node-mongo-cli/blob/develop/src/cli.js). Please do not change any other thing i.e. don't make changes to the parseArgumentsIntoOptions function, the cli function and anything inside main.js.

## Task

Error is displayed in the console when template is `undefined` or `invalid`. You are to create prompts which will solve this issue:

* Create a prompt of `type: list` with message: `please choose which project template to use`, when user does not add template as a command line argument. The prompt should list the 3 templates.
* Create a prompt of `type: list` with message: `Cli does not have template: "wrong-template-user-entered` in its template collection`, if template name provided by the user is not among the 3 template names stored in the templateCollection variable. The prompt should list the 3 templates.

#### Test cases & expected output:
**1. Run** `node-mongo` 

Prompt style:

template-1a-undefined-prompt

Expected output when `es6` is selected:

template-1b-undefined-output

📌**Note** Please do not attempt to fix the folder name here because it returns `undefined`. That is the work of whoever is assigned to the [options.folderName prompt issue](https://github.com/code-collabo/node-mongo-cli/issues/36).

**2. Run** `node-mongo cjs-folder cjs` 
Prompt style: same as above.

Expected output when `cjs` is selected:

template-2-cjsfolder-output

**3. Run** `node-mongo other java`
Prompt style:

template-3a-other-prompt

Expected output when you choose `es6` since `java` is not found in the `templateCollection`:

template-3b-other-output

#32

### LINKED ISSUE #43: Shorter command alternative for interacting with node-mongo CLI needed
#### Suggested improvement
The [command for interacting with the node-mongo CLI](https://code-collabo.gitbook.io/node-mongo-v2.3.0/#cli-command) is:
````
node-mongo
````
We don't want to remove this command, we just to add a shorter alternative:
````
nmgo
````
So the user can either choose to use the long version of the command `node-mongo` or the shorter version of the command `nmgo`.

#### Describe issue associated with the absence of this improvement
The longer `node-mongo` command is descriptive but maybe too long to type sometimes. 

#### Steps to reproduce issue associated with the absence of this improvement
Just follow the steps to use the node-mongo CLI as stated in the docs or readme

#### Example of what we want to achieve after this improvement has been added
None

#### Screenshot (example)
None

#### Modifications
None

#### Additional context
The [learning resources for the CLI in the node-mongo contribution guide](https://code-collabo.gitbook.io/node-mongo-v2.3.0/contribution-guide/node-mongo#learning-resources), will help while going about solving this issue.

#### Testing checklist
Note: Add the testing checklist below (without this note) to your pull request when submitting your fix. This will guide (you and) the contributors who will review your fix to know what is important to check or test for.
- [ ] The shorter node-mongo alternative command `nmgo` is added and works exactly as the original node-mongo command works
- [ ] The original `node-mongo` command is not removed, and still works as it used to work before
- [ ] I certify that I ran my checklist

### PR DESCRIPTION
**Pull request by Festusali**
**This pull request makes the following changes:**
* Fixes code-collabo/node-mongo#43

**General checklist**
- [x] File or folder now contains changes as specified in the issue I worked on
- [x] I have linked the issue I worked on to this pull request submitted by me

**Testing checklist**
- [x] The shorter node-mongo alternative command `nmgo` is added and works exactly as the original node-mongo command works
- [x] The original `node-mongo` command is not removed, and still works as it used to work before
- [x] I certify that I ran my checklist

Ping @code-collabo/node-mongo-cli

### CODE REVIEW

**Comment by Ifycode** [comment]
**Review comment 1 of 2:** 
PR linked to issue cleanly 👍🏽

For your pull request title: It started like this in the title "Fix #43: Shorter command alternative for interacting with node-mongo …", and ended like this in the PR description just before you linked the issue "…CLI needed".

This is preferrable and straight forward as title: "Fix: Shorter command alternative for interacting with node-mongo CLI needed"
(You can just use this as guide for subsequent pull requests - Note: this is how we like to do it here at code collabo, not like you have to adopt this for other open source communities. But this one makes particularly my work easier. That is why we are using and sticking with this).

Reasons:
- I can see the issue name straight up from the PR list if I need to find it
- Adding the issue number or id `#43` in there is a bit confusing as the pull request also has it's own number/id.

File: src/help.js [hunk 1]
```diff
@@ -48,9 +57,10 @@ folder name used already exists.
 export const notRecognised = () => {
 console.log(
 `
-Flag(s) not recognised. Use the help command below for more info:
+Flag(s) not recognised. Use any of the help command below for more info:
 
-node-mongo --help`
+node-mongo --help
+nmgo --help`
 );
```
Comment (Ifycode) [inline src/help.js:67]: **Review comment 2 of 2:**

Changes cleanly added 👍🏽 Your file changes shows this.

For the flag command info (`notRecognised` function), you can just use ideas from the changes made to the `help` function. Something like `Help command: ` and `Help command (for shorter command alternative): `

**Review by Ifycode — CHANGES_REQUESTED** [review]

File: src/help.js (hunk 1 above)
Comment (Ifycode) [reply src/help.js:67]: Just so it is consistent.

**Comment by Festusali** [comment]
> **Review comment 1 of 2:** PR linked to issue cleanly 👍🏽
> 
> For your pull request title: It started like this in the title "Fix #43: Shorter command alternative for interacting with node-mongo …", and ended like this in the PR description just before you linked the issue "…CLI needed".
> 
> This is preferrable and straight forward as title: "Fix: Shorter command alternative for interacting with node-mongo CLI needed" (You can just use this as guide for subsequent pull requests - Note: this is how we like to do it here at code collabo, not like you have to adopt this for other open source communities. But this one makes particularly my work easier. That is why we are using and sticking with this).
> 
> Reasons:
> 
> * I can see the issue name straight up from the PR list if I need to find it
> * Adding the issue number or id `#43` in there is a bit confusing as the pull request also has it's own number/id.

Thanks for the information. Consequently, I have renamed the PR and edited the accompanying comment appropriately.

File: src/help.js (hunk 1 above)
Comment (Festusali) [reply src/help.js:67]: I have made the suggested changes and push for review.

Comment (Ifycode) [reply src/help.js:67]: Thank you

**Review by Ifycode — APPROVED** [review]

**Comment by Ifycode** [comment]
@all-contributors add @Festusali for code

**Pull request merged**